dds/DCPS/RecorderImpl.cpp

```diff
@@ -228,6 +228,8 @@ void RecorderImpl::data_received(const ReceivedDataSample& sample)
 
   // we only support SAMPLE_DATA messages
   if (sample.header_.message_id_ == SAMPLE_DATA && listener_.in()) {
+    // Skip the encapsulation header
+    sample.sample_->rd_ptr(4);
```

Comment:
Are we passing if the encoding some where else? If not we shouldn't be skipping this.

Reply:
Use `sample.header_.cdr_encapsulation_`